### ISSUE TITLE
Am and Pm are mixed up

### ISSUE THREAD

**Issue by jp9856** [yuriykulikov/AlarmClock#383]
## Describe the bug
Pretty straightforward but confusing bug. Basically, Am and Pm on the app I cloned are mixed up. So, when you set an alarm for 3 am, it will actually go off at 3 pm, and vice versa. 

## To Reproduce
Describe the steps to reproduce the behavior:
Set an alarm for a couple minutes from now but put the wrong Am/Pm setting, then watch the alarm go off anyway. Then, put the correct Am/Pm setting for an alarm a couple minutes from now, and the alarm does not go off. 

I tried to debug this myself in a number of ways, and ultimately what I found was that in the AlarmCore class in the calculateNextTime() function, even though the container.hour and container.minutes values are correct, for some reason the calendar object just sets the actual time in milliseconds off by 12 hours when the get(Calendar.DAY_OF_WEEK) is called in the DaysOfWeek.getNextAlarm() helper function. It's a very confusing bug and I don't know why it's happening. There's a very simple solution of just reversing the boolean logic in TimePickerPresenter by lazy initiation of the hours variable. Essentially, just change this 

`hours == 12 && amPm == AmPm.AM -> 0
 hours == 12 && amPm == AmPm.PM -> 12
 amPm == AmPm.PM -> hours + 12` 

to this

`hours == 12 && amPm == AmPm.PM -> 0
 hours == 12 && amPm == AmPm.AM -> 12
 amPm == AmPm.AM -> hours + 12` 

This would probably work, but would also be more like a 2 wrongs making a right type of solution. I'm happy to implement this change and do a PR, but I'm more interested in why the Calendar object you are passing from the AlarmCore is behaving this way in the first place. Is Calendar no longer being maintained properly? Should we be using something else? 

Let me know if you are having this error as well.

**Comment by yuriykulikov** [comment]
Thank you for reporting this. Kind of explains why I have much less users in am/pm countries compared to 24-hour countries.

I am not used to am/pm time format because 24 hour format was used in all countries I have lived in🤔

I will look into that bug. Most likely Calendar instance is created the the Locale which implies some conversions. Calendar class is a part of Android, so it is not impossible that it was changed by the vendor, but it is highly unlikely. Most likely it is a big in my code because of my misunderstanding of the API.
